tap_s3_csv/s3.py

```diff
@@ -70,16 +71,20 @@ def setup_aws_client(config: Dict) -> None:
         boto3.setup_default_session(profile_name=aws_profile)
 
 
-def get_sampled_schema_for_table(config: Dict, table_spec: Dict) -> Dict:
+def get_sampled_schema_for_table(config: Dict, state: Dict, table_spec: Dict) -> Dict:
     """
     Detects json schema using a sample of table/stream data
+    :param state: current state
     :param config: Tap config
     :param table_spec: tables specs
     :return: detected schema
     """
     LOGGER.info('Sampling records to determine table schema.')
 
-    s3_files_gen = get_input_files_for_table(config, table_spec)
+    modified_since = utils.strptime_with_tz(get_bookmark(state, table_spec['table_name'], 'modified_since') or
```

Comment:
Discovery mode doesn't use state, the date we need to use would be `start_date` from config only.